cluster-autoscaler/processors/nodegroups/nodegroup_management_processor.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodegroups
+
+import (
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/errors"
+)
+
+// NodeGroupManagementProcessor is responsible for creating/deleting node groups.
+type NodeGroupManagementProcessor interface {
```

Comment:
Can we just call it a NodeGroupManager? I don't think it's a 'processor' in the sense other processors are:
1. You expect a side effect (creating NodeGroup). Other processors just modify input or (in case of ScaleUpStatusProcessor) have side effects that don't impact CA operations, like reporting status to users.
1. Empty implementation (or implementation output) is fine for processors. It's not fine here. It's pretty obvious that `CreateNodeGroupIfNotExist` implementation that does nothing is just plain wrong.
1. You can't have multiple implementations that you run consecutively or that call each other. 

I would rename it to manager and add a DeleteNodeGroup method to it. I would also consider pushing it through AutoscalingContext instead of AutoscalingProcessors, though I can probably accept adding it to AutoscalingProcessors for practical reasons.

Reply:
done